src/main/kotlin/skytils/skytilsmod/features/impl/handlers/SpiritLeap.kt

```diff
@@ -119,6 +125,13 @@ class SpiritLeap : PersistentSave(File(Skytils.modDir, "spiritleap.json")) {
         }
     }
 
+    @SubscribeEvent
+    fun onChatReceived(event: ClientChatReceivedEvent) {
+        if (!Skytils.config.highlightDoorOpener || !Utils.inDungeons || event.type == 2.toByte()) return
+        doorOpener = if (event.message.formattedText == bloodOpenedString) null
+        else (doorOpenedPattern.find(event.message.unformattedText)?.groups?.get("name")?.value ?: return)
```

Comment:
not too sure if this works

Reply:
The only thing that I haven't tested is an unranked player, everything else works fine. Theoretically unranked players should work, but I can't be 100% certain without actually testing it

Reply:
nons don't matter

Reply:
seems like it would work though